src/openforms/authentication/tests/test_signals.py

```diff
@@ -85,6 +86,68 @@ def test_attribute_set_for_demo_plugin_with_staff_user(self):
         instance.refresh_from_db()
         self.assertEqual(instance.auth_info.value, "123")
 
+    def test_ignore_set_attribute_if_anonymous(self):
+        register = Registry()
+        register("plugin1")(RequiresAdminPlugin)
+        instance = SubmissionFactory.create()
+        request = factory.get("/foo")
+        request.user = StaffUserFactory.create()
+        request.session = {
+            FORM_AUTH_SESSION_KEY: {
+                "plugin": "plugin1",
+                "attribute": RequiresAdminPlugin.provides_auth,
+                "value": "123",
+            }
+        }
+
+        with mock_register(register):
+            set_auth_attribute_on_session(
+                sender=None, instance=instance, request=request, anonymous=True
+            )
+
+        instance.refresh_from_db()
+        with self.assertRaises(Submission.auth_info.RelatedObjectDoesNotExist):
+            instance.auth_info
+
+    @tag("gh-4199")
+    def test_two_submissions_within_same_session(self):
+        """
+        If a user starts a Submission on a Form with authentication and a second
+        Submission on a form without authentication, the second Submission should not have
+        any authentication details associated with it
+        """
+        register = Registry()
+        register("plugin1")(RequiresAdminPlugin)
+        submission_with_auth = SubmissionFactory.create()
+        request = factory.get("/foo")
+        request.user = StaffUserFactory.create()
+        request.session = {
+            FORM_AUTH_SESSION_KEY: {
+                "plugin": "plugin1",
+                "attribute": RequiresAdminPlugin.provides_auth,
+                "value": "123",
+            }
+        }
+
+        with mock_register(register):
+            set_auth_attribute_on_session(
+                sender=None, instance=submission_with_auth, request=request
+            )
+
+        submission_with_auth.refresh_from_db()
+        self.assertEqual(submission_with_auth.auth_info.value, "123")
+
+        submission_no_auth = SubmissionFactory.create()
+
+        with mock_register(register):
+            set_auth_attribute_on_session(
+                sender=None, instance=submission_no_auth, request=request
+            )
```

Comment:
is the absence of `anonymous=True` deliberate here?

Reply:
I made this test before I added the `anonymous` flag I think, I'll add it when I move these tests to the `start_submission` tests